tools/optimizer/simple_ast.cpp

```diff
@@ -93,7 +93,9 @@ struct StackedStack { // a stack, on the stack
         memcpy(storage, old, sizeof(T)*used);
         alloced = true;
       } else {
-        storage = (T*)realloc(storage, sizeof(T)*available);
+        T *newStorage = (T*)realloc(storage, sizeof(T)*available);
+        assert(newStorage || !storage);
```

Comment:
it's a slight shame to clutter up the code just to avoid a static analyzer false positive. but if there's no easy better way, i'm ok with it.